api/src/database/db-utils.ts

```diff
@@ -81,3 +102,59 @@ export const getZodQueryResult = <T extends z.Schema>(zodQueryResultRow: T) =>
       })
     )
   });
+
+/**
+ * Converts a type specific keycloak user information object with type specific properties into a new object with
+ * generic properties.
+ *
+ * @param {KeycloakUserInformation} keycloakUserInformation
+ * @return {*}  {(GenericizedKeycloakUserInformation | null)}
+ */
+export const getGenericizedKeycloakUserInformation = (
+  keycloakUserInformation: KeycloakUserInformation
+): GenericizedKeycloakUserInformation | null => {
+  let data: GenericizedKeycloakUserInformation | null;
+
+  if (isDatabaseUserInformation(keycloakUserInformation)) {
+    // Don't patch internal database user records
+    return null;
+  }
+
+  // We don't yet know at this point what kind of token was used (idir vs bceid basic, etc).
+  // Determine which type it is, and parse the information into a generic structure that is supported by the
+  // database patch function
+  if (isIdirUserInformation(keycloakUserInformation)) {
+    data = {
+      user_guid: keycloakUserInformation.idir_user_guid,
+      user_identifier: keycloakUserInformation.idir_username,
+      user_identity_source: SYSTEM_IDENTITY_SOURCE.IDIR,
+      display_name: keycloakUserInformation.display_name,
+      email: keycloakUserInformation.email,
+      given_name: keycloakUserInformation.given_name,
+      family_name: keycloakUserInformation.family_name
+    };
+  } else if (isBceidBusinessUserInformation(keycloakUserInformation)) {
+    data = {
+      user_guid: keycloakUserInformation.bceid_user_guid,
+      user_identifier: keycloakUserInformation.bceid_username,
+      user_identity_source: SYSTEM_IDENTITY_SOURCE.BCEID_BUSINESS,
+      display_name: keycloakUserInformation.display_name,
+      email: keycloakUserInformation.email,
+      given_name: keycloakUserInformation.given_name,
+      family_name: keycloakUserInformation.family_name,
+      agency: keycloakUserInformation.bceid_business_name
+    };
+  } else {
```

Comment:
Maybe we should make this an else-if branch and chec kthat it's actually a BCeID Basic account? Because if it's not (for whatever reason), then we probably want to just return `null`, right?

Reply:
Discussed: Zod schema check asserts that the account is one of four types, so this check is not neccessary.